requests_unixsocket/__init__.py

```diff
@@ -31,5 +33,34 @@ def __enter__(self):
 
     def __exit__(self, *args):
         requests = self._get_global_requests_module()
-        requests.get = self.orig_requests_get
-        requests.request = self.orig_requests_request
+        for m in self.methods:
+            requests.__dict__[m] = self.orig_methods[m]
+
+# These are the same methods defined for the global requests object
+def request(method, url, **kwargs):
+    session = Session()
+    return session.request(method=method, url=url, **kwargs)
+
+def get(url, **kwargs):
+    kwargs.setdefault('allow_redirects', True)
+    return request('get', url, **kwargs)
+
+def head(url, **kwargs):
+    kwargs.setdefault('allow_redirects', False)
+    return request('head', url, **kwargs)
+
+def post(url, **kwargs):
+    return request('post', url, data=data, **kwargs)
+
+def patch(url, **kwargs):
+    return request('patch', url,  data=data, **kwargs)
```

Comment:
The `data` variable is not defined -- presumably, this is not needed, because it will be in `**kwargs`.

```
$ tox -e flake8 | grep 'undefined'
./requests_unixsocket/__init__.py:53:38: F821 undefined name 'data'
./requests_unixsocket/__init__.py:56:40: F821 undefined name 'data'
./requests_unixsocket/__init__.py:59:37: F821 undefined name 'data'
```